sdk/python/test_data/components/container_component_with_list_of_artifacts.py

```diff
@@ -0,0 +1,18 @@
+from typing import List
+
+from kfp import dsl
+from kfp.dsl import Artifact
+from kfp.dsl import Input
+
+
+@dsl.container_component
+def comp_with_artifact_list(input_list: Input[List[Artifact]]):
+    return dsl.ContainerSpec(
+        image='alpine', command=[input_list], args=[input_list])
```

Comment:
I think we should have an example that's executable to show case how the container can consume the list of artifacts, like printing out the artifact names, etc. Without that, the input is an opaque block and it's unclear to users how to actually use list of artifacts in container component.

Reply:
Agreed. For this PR, I've opted to remove read/write tests altogether since lists of artifacts are unusable until ParallelFor fan-in is implemented.

I will add real-world test cases in alongside ParallelFor fan-in (https://github.com/kubeflow/pipelines/pull/8808).

Reply:
SGTM